src/main/java/ch/njol/skript/expressions/ExprArrowDamage.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * This file is part of Skript.
+ * <p>
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ * <p>
+ * <p>
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.entity.AbstractArrow;
+import org.bukkit.entity.Arrow;
+import org.bukkit.entity.Projectile;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Arrow Damage")
+@Description("An arrow's base damage. ")
+@Examples({"on shoot:",
+	"\tevent-projectile is an arrow",
+	"\tset arrow damage of event-projectile to 0"})
+@Since("INSERT VERSION")
+public class ExprArrowDamage extends SimplePropertyExpression<Projectile, Number> {
+	
+	final static boolean abstractArrowExists = Skript.classExists("org.bukkit.entity.AbstractArrow");
+	
+	static {
+		if (abstractArrowExists || Skript.methodExists(Arrow.class, "getDamage"))
+			register(ExprArrowDamage.class, Number.class, "[the] arrow damage", "projectiles");
+	}
+	
+	@Nullable
+	@Override
+	public Number convert(Projectile arrow) {
+		if (abstractArrowExists)
+			return arrow instanceof AbstractArrow ? ((AbstractArrow) arrow).getDamage() : null;
+		return arrow instanceof Arrow ? ((Arrow) arrow).getDamage() : null;
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return (mode == ChangeMode.DELETE || mode == ChangeMode.REMOVE_ALL) ? null : CollectionUtils.array(Number.class);
+	}
+	
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		double strength = delta != null ? Math.max(((Number) delta[0]).doubleValue(), 0) : 0;
+		switch (mode) {
+			case REMOVE:
+				if (abstractArrowExists)
```

Comment:
maybe it would make more sense to make a getDamage and setDamage method in `ch.njol.skript.bukkitutil.ProjectileUtils` so you don't have to repeat this check and the logic every time

Reply:
But not all projectiles have a getDamage and setDamage event

Reply:
yeah but the class isn't specifically for things that applies to all projectiles, it's just a util class for things related to projectiles

Reply:
Hey, I'm going through the pull requests and updating them, anyways would I do something like this or do you think it'd be better to do it in another way?

```java
	static {
		try {
			getDamage = AbstractArrow.class.getMethod("getDamage");
			setDamage= AbstractArrow.class.getMethod("setDamage", doule.class);
		} catch (final NoSuchMethodException e) {
			try {
				getDamage = Arrow.class.getMethod("getDamage");
				setDamage= Arrow.class.getMethod("setDamage", doule.class);
			} catch (final NoSuchMethodException e2) {
				Skript.outdatedError(e);
			}
		} catch (final SecurityException e) {
			Skript.exception(e, "security manager present");
		}
	}

```

Reply:
you could do something like this:
```
private final static boolean ABSTRACT_ARROW_EXISTS = Skript.classExists("org.bukkit.entity.AbstractArrow");

public static void setDamage(Projectile arrow, double damage) {
    if (ABSTRACT_ARROW_EXISTS && arrow instanceof AbstractArrow) {
        ((AbstractArrow) arrow).setDamage(damage);
    } else if (arrow instanceof Arrow) {
        ((Arrow) arrow).setDamage(damage);
    }
}
```